server/storage/filtering/filtering.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filtering
```

Comment:
Similarly to the dao package, I'd feel better if this was just included in the storage package - I think the argument for merging them is that these are all part of our storage subsystem (I see this is subpackage, but it's small). I don't easily see situations where we'd want to use one of these parts without another, so the partitioning seems like it's just an organizational convenience that complicates our package namespace. (I may have written code inconsistent with this before, this is just my reaction to what I'm seeing now, happy to rethink this)

Reply:
I agree with your concerns about the package namespace complexity in the current state (`server/dao` depends on `server/storage` and `server/storage/filtering`), but I think it's important to look at this with the end result in mind (`server/storage` will depend on `server/storage/filtering`).

It is a small package and it's only used by its parent, but I don't see why that's problematic. Personally I think it's beneficial to encapsulate filtering names in their own namespace because the alternative is to push that separation into the names themselves (e.g. `filtering.String` becomes `storage.FilterString`). Could you elaborate on your concerns?

Reply:
OK, as above, SGTM. I think the "filtering" namespace clarification is sufficient justification for this. Testing this independently would probably also make sense.